jsonrpc/filter_manager.go

```diff
@@ -818,14 +809,23 @@ func (t *timeHeapImpl) Pop() interface{} {
 // of the stream at any point
 type blockStream struct {
 	lock sync.Mutex
-	head *headElem
+	head atomic.Value
+}
+
+func newBlockStream(head *block) *blockStream {
+	b := &blockStream{}
+	b.head.Store(&headElem{header: head})
+
+	return b
 }
 
 func (b *blockStream) Head() *headElem {
```

Comment:
Nitpick, unrelated to this PR:
`Head` can be unexported, as it is used only by `jsonrpc` package.